packages/contracts/src/core/plugin/proposal/Proposal.sol

```diff
@@ -4,25 +4,72 @@ pragma solidity 0.8.17;
+abstract contract Proposal is IProposal {
     using Counters for Counters.Counter;
 
     /// @notice The incremental ID for proposals and executions.
     Counters.Counter private proposalCounter;
 
-    /// @inheritdoc ProposalBase
+    /// @inheritdoc IProposal
     function proposalCount() public view override returns (uint256) {
         return proposalCounter.current();
     }
 
-    /// @inheritdoc ProposalBase
-    function createProposalId() internal override returns (uint256 proposalId) {
+    /// @notice Creates a proposal ID.
+    /// @return proposalId The proposal ID.
+    function _createProposalId() internal virtual returns (uint256 proposalId) {
         proposalId = proposalCount();
         proposalCounter.increment();
     }
+
+    /// @notice Internal function to create a proposal.
+    /// @param _metadata The the proposal metadata.
+    /// @param _startDate The start date of the proposal in seconds.
+    /// @param _endDate The end date of the proposal in seconds.
+    /// @param _allowFailureMap A bitmap allowing the proposal to succeed, even if individual actions might revert. If the bit at index `i` is 1, the proposal succeeds even if the `i`th action reverts. A failure map value of 0 requires every action to not revert.
+    /// @param _actions The actions that will be executed after the proposal passes.
+    /// @return proposalId The ID of the proposal.
+    function _createProposal(
+        address _creator,
+        bytes calldata _metadata,
+        uint64 _startDate,
+        uint64 _endDate,
+        IDAO.Action[] calldata _actions,
+        uint256 _allowFailureMap
+    ) internal virtual returns (uint256 proposalId) {
+        proposalId = _createProposalId();
+
+        emit ProposalCreated({
+            proposalId: proposalId,
+            creator: _creator,
+            metadata: _metadata,
+            startDate: _startDate,
+            endDate: _endDate,
+            actions: _actions,
+            allowFailureMap: _allowFailureMap
+        });
+    }
+
+    /// @notice Internal function to execute a proposal.
+    /// @param _proposalId The ID of the proposal to be executed.
+    /// @param _actions The array of actions to be executed.
+    /// @param _allowFailureMap A bitmap allowing the proposal to succeed, even if individual actions might revert. If the bit at index `i` is 1, the proposal succeeds even if the `i`th action reverts. A failure map value of 0 requires every action to not revert.
+    function _executeProposal(
+        IDAO _dao,
+        uint256 _proposalId,
+        IDAO.Action[] memory _actions,
+        uint256 _allowFailureMap
+    ) internal virtual {
+        (bytes[] memory execResults, ) = _dao.execute(
+            bytes32(_proposalId),
+            _actions,
+            _allowFailureMap
+        );
+        emit ProposalExecuted({proposalId: _proposalId, execResults: execResults});
```

Comment:
If we emit `execResults`, why not emit `failureMap` ? It's either we emit both, or none of them. Sorry that I realized this now.

Reply:
Yes, it should be emitted. Should we add a `failed: Boolean!` to
```graphql
type Action @entity {
  id: ID!
  to: Bytes!
  value: BigInt!
  data: Bytes!
  dao: Dao!
  proposal: Proposal!
  execResult: Bytes
}
```
?

Reply:
I think this should be a separate subgraph task, probably for you @novaknole. @brickpop

Reply:
If the fix for what @novaknole proposes is trivial here, I would add it here and save some overhead

Reply:
The fix by Giorgi changes the subgraph and SDK, so we should do it in a separate PR. 
Please create a task for it @brickpop.